### DISCUSSION TITLE
Welcome to onedrive-cf-index-ng Discussions!

### DISCUSSION THREAD

**Discussion by lyc8503** [lyc8503/onedrive-cf-index-ng#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Izzy2401** [comment]
Hi, is it possible to configure the index to use the root of a sharepoint site instead?

**Reply by lyc8503** [reply]
It is possible, you can refer to https://github.com/spencerwooo/onedrive-vercel-index/issues/817

**Reply by Izzy2401** [reply]
Thanks. Been trying this but no success as yet. Will keep at it. All the best and many thanks for this project.

**Reply by Izzy2401** [reply]
For anyone who may not have been able to get this project to work with Sharepoint, these are the steps I took in order to get it to work. 

1) Follow lyc8503's guide and register the token
2) Get your Sharepoint site ID using: https://**_TENENT_**.sharepoint.com/sites/**_SITEHERE_**/_api/site/id
3) Change the drive API to https ://graph.microsoft.com/v1.0/sites/**_SITEID_**/drive in config/api.config.js <-----remove space after https

**Comment by Izzy2401** [comment]
If I would like to add a dropdown/navigation menu in order to swap between multiple indexs, can you recommend where this should be done? Thanks for all of the work you have put into this project.

**Reply by lyc8503** [reply]
Do you mean multiple Microsoft accounts? That requires many changes. You may just host 2 instances of this project and put links to each other.

**Reply by Izzy2401** [reply]
A dropdown menu similar to the one seen here. https://dark-demo.dudoo.workers.dev/0:/
It allows you to swap between multiple onedrive/sharepoint drives. Hosting multiple instances isn't a problem, its just having an easy way to navigate between them. Hope this clarifies. If it requires making many changes, where would you suggest I place the links?

**Reply by lyc8503** [reply]
You may add external links directly in config file `site.config.js`. That's the easiest way to do that. If you want more customization (e.g. dropdown menu), you can take a look at [this file](https://github.com/lyc8503/onedrive-cf-index-ng/blob/main/src/components/Navbar.tsx) and make your own changes in it. This requires some coding skills in frontend though.

**Reply by Izzy2401** [reply]
Thank you. Hopefully I can provide an update for others who may want to do so.
